### ISSUE TITLE
[BUG] AttributeError: 'DefaultForecasterConfig' object has no attribute 'granularity'

### ISSUE THREAD

**Issue by swarmt** [salesforce/Merlion#53]
Running 0_ForecastIntro.ipynb

When training default model throws an Exception
```
AttributeError: 'DefaultForecasterConfig' object has no attribute 'granularity'
```

**To Reproduce**
```python
from merlion.models.defaults import DefaultForecasterConfig, DefaultForecaster
model = DefaultForecaster(DefaultForecasterConfig())
model.train(train_data=train_data)
```

 - OS: Colab
 - Merlion Version 1.0.2 and 1.1.0

**Comment by sirmick** [comment]
Yeah this is a bug for sure. I managed to work around this by using model.config.granularity = None after creating the model, but before using "train"

**Comment by swarmt** [comment]
```python
from merlion.models.defaults import DefaultForecasterConfig, DefaultForecaster
model = DefaultForecaster(DefaultForecasterConfig())
model.config.granularity = None
model.train(train_data=train_data)
test_pred, test_err = model.forecast(time_stamps=test_data.time_stamps)
```
```
AttributeError: 'DefaultForecasterConfig' object has no attribute 'max_forecast_steps'
```

Does not appear to be any unit tests for DefaultForecaster unless I am mistaken.

**Comment by swarmt** [comment]
I can work around it by not using the DefaultForcaster 
```python
from merlion.models.factory import ModelFactory

granularity = None
max_forecast_steps = 5
target_seq_index = None

transform_dict = dict(name="TemporalResample", granularity=granularity)
kwargs = dict(
    transform=transform_dict, max_forecast_steps=max_forecast_steps, target_seq_index=target_seq_index
)

model = ModelFactory.create("ETS", damped_trend=True, **kwargs)
```

**Comment by aadyotb** [comment]
Thanks for surfacing this issue. I'll work on a fix soon.

**Comment by aadyotb** [comment]
The fix is implemented in in #57.
